tests/is-winner/pass_afonso.js

```diff
@@ -0,0 +1,42 @@
+const isWinner = (candidate) =>
+  winners()
+    .getWinner(candidate)
+
+    .catch((err) => Promise.reject(candidate + ' never was a winner'))
+
+    .then((countryInfo) =>
+      countryInfo.continent === 'Europe'
+        ? Promise.resolve(countryInfo.id)
+        : Promise.reject(
+            candidate +
+              ' is not what we are looking for because of the continent'
+          )
+    )
+
+    .then(async (countryId) => {
+      let arrInfo = await winners().getResults(countryId)
+      return arrInfo.length >= 3
+        ? Promise.resolve(
+            candidate +
+              ' won the FIFA World Cup in ' +
+              (arrInfo.length === 1
+                ? arrInfo[0].year
+                : arrInfo.reduce((acc, obj, i) =>
+                    i === 1 ? acc.year + ', ' + obj.year : acc + ', ' + obj.year
+                  )) +
+              ' winning by ' +
+              (arrInfo.length === 1
+                ? arrInfo[0].result
+                : arrInfo.reduce((acc, obj, i) =>
+                    i === 1
+                      ? acc.result + ', ' + obj.result
+                      : acc + ', ' + obj.result
+                  ))
+          )
+        : Promise.reject(
+            candidate +
+              ' is not what we are looking for because of the number of times it was champion'
+          )
+    })
+
+    .catch((d) => d)
```

Comment:
`.catch((d) => d)` is not something you often want, I do it for tests but this would catch all errors and might give you surprising results